initial_setup/__init__.py

```diff
@@ -0,0 +1,289 @@
+            raise InitialSetupError
+
+        if external_reconfig:
+            # set the reconfig flag in kickstart so that
+            # relevant spokes show up
+            self.data.firstboot.firstboot = FIRSTBOOT_RECONFIG
+
+    def _setup_locale(self):
+        log.debug("setting up locale")
+        # Normalize the locale environment variables
+        if self.data.lang.seen:
+            locale_arg = self.data.lang.lang
+        else:
+            locale_arg = None
+        setup_locale_environment(locale_arg, prefer_environment=True)
+        setup_locale(os.environ['LANG'], text_mode=not self.gui_mode)
+
+    def _apply(self):
+        # Do not execute sections that were part of the original
+        # anaconda kickstart file (== have .seen flag set)
+
+        sections = [self.data.keyboard, self.data.lang, self.data.timezone]
+
+        # data.selinux
+        # data.firewall
+
+        log.info("executing kickstart")
+        for section in sections:
+            section_msg = "%s on line %d" % (repr(section), section.lineno)
+            if section.seen:
+                log.debug("skipping %s", section_msg)
+                continue
+            log.debug("executing %s", section_msg)
+            section.execute(None, self.data, None)
+
+        # Prepare the user database tools
+        u = Users()
+
+        sections = [self.data.group, self.data.user, self.data.rootpw]
+        for section in sections:
+            section_msg = "%s on line %d" % (repr(section), section.lineno)
+            if section.seen:
+                log.debug("skipping %s", section_msg)
+                continue
+            log.debug("executing %s", section_msg)
+            section.execute(None, self.data, None, u)
+
+        # Configure all addons
+        log.info("executing addons")
+        self.data.addons.execute(None, self.data, None, u)
+
+        if external_reconfig:
+            # prevent the reconfig flag from being written out,
+            # to prevent the reconfig mode from being enabled
+            # without the /etc/reconfigSys file being present
+            self.data.firstboot.firstboot = None
+
+        # Write the kickstart data to file
+        log.info("writing the Initial Setup kickstart file %s", OUTPUT_KICKSTART_PATH)
+        with open(OUTPUT_KICKSTART_PATH, "w") as f:
+            f.write(str(self.data))
+        log.info("finished writing the Initial Setup kickstart file")
+
+        # Remove the reconfig file, if any - otherwise the reconfig mode
+        # would start again next time the Initial Setup service is enabled
+        if external_reconfig and os.path.exists(RECONFIG_FILE):
+            log.debug("removing the reconfig file")
+            os.remove(RECONFIG_FILE)
+            log.debug("the reconfig file has been removed")
+
+    def run(self):
+        """Run Initial setup
+
+        :param bool gui_mode: if GUI should be used (TUI is the default)
```

Comment:
I don't know if using param in doctype is good idea when this is parameter passed in by self.gui_mode.